lago/vm.py

```diff
@@ -223,8 +223,13 @@ def _libvirt_name(self):
         return self.vm.virt_env.prefixed_name(self.vm.name())
 
     def _libvirt_xml(self):
-        with open(_path_to_xml('dom_template.xml')) as xml_fd:
-            dom_raw_xml = xml_fd.read()
+        metadata = self.vm.metadata
+        if metadata and 'custom-xml' in metadata:
+            with open(metadata['custom-xml']) as xml_fd:
+                dom_raw_xml = xml_fd.read()
+        else:
+            with open(_path_to_xml('dom_template.xml')) as xml_fd:
+                dom_raw_xml = xml_fd.read()
```

Comment:
Maybe move this to it's own function, something like self.vm.get_domxml that does the fallback to the default template (or if you want to be fancy, using the python getters and setters for properties, so self.vm.domxml does it magically).